packages/plugin-node/src/services/awsS3.ts

```diff
@@ -117,7 +125,14 @@ export class AwsS3Service extends Service implements IAwsS3Service {
 
             // If not using signed URL, return public access URL
             if (!useSignedUrl) {
-                result.url = `https://${this.bucket}.s3.${process.env.AWS_REGION}.amazonaws.com/${fileName}`;
+                // Use the endpoint directly (e.g., local S3) if a custom value was set
+                if (this.s3Client.config.endpoint) {
+                    const endpoint = await this.s3Client.config.endpoint();
+                    const port = endpoint.port ? `:${endpoint.port}` : "";
```

Comment:
only `port` is optional; the other fields are non-optional so can be used directly